Automattic/woocommerce-payments
# Created Locale class and moved locale functions into it.

Fixes: #2407

#### Changes proposed in this Pull Request

* Create `Locale` class to give locale based functions and data to live.
* Moved `load_locale_data` from `FrontendCurrencies` to `Locale`, and updated calls within `FrontendCurrencies` accordingly.
* Moved `get_currency_format` from `Utils` to `Locale`, updated calls to it accordingly.
* `FrontendCurrencies`  now has a dependency on `Locale` and not `Utils`.
* Added and updated tests.

#### Testing instructions

* Set your store default currency to USD.
* Enable EUR in your store under WooCommerce > Settings > Multi-Currency.
* View the shop page and switch from USD to EUR
* The formatting should (still) change from $1,000.00 to 1.000,00 €.
* Tests should pass.

-------------------
